### DISCUSSION TITLE
blog/2022/install-github-from-pyproject/

### DISCUSSION THREAD

**Discussion by giscus** [choldgraf/choldgraf.github.io#73] [Blog comments]
# blog/2022/install-github-from-pyproject/

This is a short post to demonstrate how to install packages directly from GitHub with pyprojects.toml or requirements.txt, including custom branches and commits. It will focus on pyprojects.toml be...

https://chrisholdgraf.com/blog/2022/install-github-from-pyproject/

**Comment by JannKleen** [comment]
Hi Chris, thanks for writing this! The way you are specifying the branch / commit is incorrect though. Your example should look like this: `"<packagename>@git+<url-to-repo>@<branch or hash>#egg=<packagename>"`.

**Reply by alanbuxton** [reply]
@chris - please do update your doc with the corrected syntax that JannKleen mentions. It took me quite a bit of digging around to get to the corrected syntax. TBH the `#egg=` piece seems redundant, at least I didn't need to use it. The syntax shown here worked for me: https://github.com/alanbuxton/django-neomodel/blob/d524f028dcb84b779d35f9ea0ba405b55e756764/pyproject.toml#L36

**Comment by chribro88** [comment]
@alanbuxton I had to add `.git` to get it to work
```
dependencies = [
<packagename>@git+<url-to-repo>.git@<branch or hash>
```

**Reply by chribro88** [reply]
I spoke to soon... adding `.git` to the repo url isn't really needed, I had conda package mess up the git dep when i was trying to figure it out..

these formats seem to work which is what @alanbuxton mentioned
```
<packagename>@git+<url-to-repo>@<branch or hash>
<packagename>@git+<url-to-repo>@<branch or hash>#egg=<packagename>
```

this post seems to explain #egg a bit more [https://stackoverflow.com/a/55526482/7573470](https://stackoverflow.com/a/55526482/7573470)

and here also [https://github.com/pypa/pip/issues/11617](https://github.com/pypa/pip/issues/11617)
